src/Raven.Server/Documents/Indexes/MapReduce/Auto/AutoMapReduceIndexDefinition.cs

```diff
@@ -133,6 +133,19 @@ protected void PersistGroupByFields(JsonOperationContext context, AbstractBlitta
                 first = false;
             }
             writer.WriteEndArray();
+
+            writer.WritePropertyName((nameof(GroupByFieldNames)));
+            writer.WriteStartArray();
+            first = true;
+            foreach (var field in GroupByFieldNames)
```

Comment:
do we need that? index is getting compiled anyway so you have the fields there?

Reply:
I'm looking for the field names in their original order.
When we export and import, we won't have it anymore.
I'll add a test.

Reply:
Added a test